data/data/azure/cluster/dns/dns.tf

```diff
@@ -22,7 +22,7 @@ resource "azurerm_private_dns_zone_virtual_network_link" "network" {
   private_dns_zone_name = azurerm_private_dns_zone.private.name
   virtual_network_id    = var.virtual_network_id
 
-  depends_on = [time_sleep.wait_30_seconds]
+  depends_on = [time_sleep.wait_30_seconds, var.virtual_network_id]
```

Comment:
Aren't we failing on the private_dns_zone or is there something pointing at the vnet 404'ing too? From what I can tell looking at logs, the virtual network gets created in a much earlier vnet stage which happens before the cluster stage.

Reply:
But if this makes it through CI maybe it's fine to have the explicit ordering here.

One other thought, what if you moved azurerm_private_dns_zone.private to the vnet stage? That'd give it a much longer time for Azure caches to get in sync.

Reply:
Yup. I'm barking up the wrong tree. I see now that the private zone does not exist when this fails.

Reply:
> Yup. I'm barking up the wrong tree.

I'm glad to have more barkers :) I've exhausted my terraform abilities in trying to make this work. :)